core/src/test/java/com/linecorp/armeria/server/websocket/WebSocketServiceHttp2RequestCompleteTest.java

```diff
@@ -109,15 +107,13 @@ void closeAfterTimeoutWhenCloseFrameIsNotSent() throws InterruptedException {
         final HttpRequestWriter requestWriter =
                 HttpRequest.streaming(webSocketUpgradeHeaders("/2000MillisTimeout"));
         final BodySubscriber bodySubscriber = new BodySubscriber();
-        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
-            final SplitHttpResponse split = client.execute(requestWriter).split();
-            split.headers().thenAccept(headers -> {
-                assertThat(headers.status()).isSameAs(HttpStatus.OK);
-                // Update the request to prevent the request from being completed.
-                captor.get().updateRequest(HttpRequest.of(HttpMethod.GET, "/"));
-            });
-            split.body().subscribe(bodySubscriber);
-        }
+        final SplitHttpResponse split = WebClient.builder(client.uri())
+                                                 // Not to send the close frame from the client.
+                                                 .requestAutoAbortDelayMillis(Long.MAX_VALUE)
+                                                 .build()
+                                                 .execute(requestWriter).split();
+        split.headers().thenAccept(headers -> assertThat(headers.status()).isSameAs(HttpStatus.OK));
```

Comment:
nit; not that it matters but I believe assertions that aren't thrown from the main junit test thread won't be propagated (i.e. won't count towards test failure)

Reply:
That is true. Updated. 😉